contracts/OlasMech.sol

```diff
@@ -308,25 +276,34 @@ abstract contract OlasMech is Mech, IErrorsMech, ImmutableStorage {
         _locked = 1;
     }
 
+    /// @dev Sets up a mech.
+    /// @param initParams Mech initial parameters.
     function setUp(bytes memory initParams) public override {
-        require(readImmutable().length == 0, "Already initialized");
+        if (readImmutable().length != 0) {
+            revert AlreadyInitialized();
+        }
         writeImmutable(initParams);
     }
 
-    function token() public view returns (address serviceRegistry) {
+    /// @dev Gets mech token (service registry) address.
+    /// @return serviceRegistry Service registry address.
+    function token() external view returns (address serviceRegistry) {
+        // Get service registry
         serviceRegistry = abi.decode(readImmutable(), (address));
     }
 
-    function tokenId() public view returns (uint256) {
-        (, uint256 serviceId) = abi.decode(readImmutable(), (address, uint256));
-        return serviceId;
+    /// @dev Gets mech token Id (service Id).
+    /// @return serviceId Service Id.
+    function tokenId() external view returns (uint256 serviceId) {
+        // Get service Id
+        (, serviceId) = abi.decode(readImmutable(), (address, uint256));
     }
 
-    function isOperator(address signer) public view override returns (bool) {
-        (address serviceRegistry, uint256 serviceId) = abi.decode(
-            readImmutable(),
-            (address, uint256)
-        );
+    /// @dev Gets mech operator (service multisig).
+    /// @return Service multisig address.
+    function getOperator() public view returns (address) {
+        // Get service registry and service Id
+        (address serviceRegistry, uint256 serviceId) = abi.decode(readImmutable(), (address, uint256));
 
         (, address multisig, , , , , IServiceRegistry.ServiceState state) =
```

Comment:
This is cleaner now, also there must be a possibility to directly get the multisig without doing all the service registry mess.